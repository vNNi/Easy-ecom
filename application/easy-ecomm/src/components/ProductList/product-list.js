import { Row, Col } from 'antd';
import { Container } from './product-list.styles';
import ProductCard from '../ProductCard';

const ProductList = ({ products = [], onItemClick }) => {
  return (
    <Container>
      <Row gutter={{ xl: 16, lg: 12, md: 10 }} justify="center">
        {products?.map(product => {
          return (
            <Col
              xl={{ span: 5, md: 3, sm: 2 }}
              style={{ display: 'flex', justifyContent: 'center' }}
            >
              <ProductCard {...product} onClick={onItemClick} />
            </Col>
          );
        })}
      </Row>
    </Container>
  );
};

export default ProductList;
