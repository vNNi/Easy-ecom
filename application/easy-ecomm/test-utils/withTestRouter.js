import React from 'react';
import { RouterContext } from 'next/dist/next-server/lib/router-context';

export default function withTestRouter(tree, router = {}) {
  const {
    route = '',
    pathname = '',
    pagename = '',
    pagetype = '',
    query = {},
    asPath = '',
    push = async () => true,
    replace = async () => true,
    reload = () => null,
    back = () => null,
    prefetch = async () => undefined,
    beforePopState = () => null,
    isFallback = false,
    events = {
      emit: () => null,
      off: () => null,
      on: () => null,
    },
  } = router;

  return (
    <RouterContext.Provider
      value={{
        asPath,
        back,
        beforePopState,
        events,
        isFallback,
        pagename,
        pagetype,
        pathname,
        prefetch,
        push,
        query,
        reload,
        replace,
        route,
      }}
    >
      {tree}
    </RouterContext.Provider>
  );
}
