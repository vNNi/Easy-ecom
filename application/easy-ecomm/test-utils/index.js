import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import '@testing-library/jest-dom/extend-expect'; // jest-dom custom assertions
import 'jest-styled-components';

import withTestRouter from './withTestRouter';
import Theme from '../src/styles/theme';

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={Theme}>{children}</ThemeProvider>;
};

export const renderWithTheme = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export { withTestRouter };

// re-export everything
export * from '@testing-library/react';
