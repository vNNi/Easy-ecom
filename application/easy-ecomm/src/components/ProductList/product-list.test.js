import ProductList from './product-list';
import { renderWithTheme } from '../../../test-utils';

describe('[Components] - product-list', () => {
  test('should render correclty', () => {
    const element = renderWithTheme(<ProductList />);
    expect(element).toMatchSnapshot();
  });
});
