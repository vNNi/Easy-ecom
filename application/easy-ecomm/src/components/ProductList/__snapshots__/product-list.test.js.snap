// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Components] - product-list should render correclty 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="product-liststyles__Container-sc-1mfw1o6-0"
      >
        <div
          class="ant-row ant-row-center"
          style="margin-left: -8px; margin-right: -8px;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="product-liststyles__Container-sc-1mfw1o6-0"
    >
      <div
        class="ant-row ant-row-center"
        style="margin-left: -8px; margin-right: -8px;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
